File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        if(word1.size()!=word2.size()) return false;
        vector<int>cnt(26,0),cnt2(26,0);
        for(int i=0;word1[i]!='\0';i++)
        {
            cnt[word1[i]-'a']++;
        }
        for(int i=0;word2[i]!='\0';i++)
        {
            if(cnt[word2[i]-'a']==0 )
                return false;
            cnt2[word2[i]-'a']++;
        }
        sort(cnt.begin(),cnt.end());
        sort(cnt2.begin(),cnt2.end());
        return cnt==cnt2;
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {boolean}
 */
var closeStrings = function(word1, word2) {
    var cnt= new Array(26).fill(0);
    var cnt2 =new Array(26).fill(0);
    for(ch of word1)
    {
        cnt[ch.charCodeAt(0)-97]+=1;
    }
    for(ch of word2)
    {
        cnt2[ch.charCodeAt(0)-97]++;
    }
    for(i=0;i<26;i++)
     { 
        if((cnt[i]==0 && cnt2[i]==0) ||(cnt[i]!=0 && cnt2[i]!=0) )  
            continue;
         else return false;
     }
    cnt.sort((a,b)=>a-b);
    cnt2.sort((a,b)=>a-b);
    for(var i=0;i<26;i++)
    {
        if(cnt[i]!=cnt2[i])   return false;
    }
    return true;
};